pkg/planner/core/casetest/tpch/tpch_test.go

```diff
@@ -0,0 +1,106 @@
+from customer, orders, lineitem
+where c_mktsegment = 'AUTOMOBILE' and c_custkey = o_custkey and l_orderkey = o_orderkey and o_orderdate < '1995-03-13' and l_shipdate > '1995-03-13'
+group by l_orderkey, o_orderdate, o_shippriority
+order by revenue desc, o_orderdate
+limit 10;`).Check(testkit.Rows(
+		"Projection_14 10.00 root  test.lineitem.l_orderkey, Column#34, test.orders.o_orderdate, test.orders.o_shippriority",
+		"└─TopN_18 10.00 root  Column#34:desc, test.orders.o_orderdate, offset:0, count:10",
+		"  └─TableReader_142 10.00 root  MppVersion: 3, data:ExchangeSender_141",
+		"    └─ExchangeSender_141 10.00 mpp[tiflash]  ExchangeType: PassThrough",
+		"      └─TopN_140 10.00 mpp[tiflash]  Column#34:desc, test.orders.o_orderdate, offset:0, count:10",
+		"        └─Projection_135 39998176.29 mpp[tiflash]  Column#34, test.orders.o_orderdate, test.orders.o_shippriority, test.lineitem.l_orderkey",
+		"          └─HashAgg_133 39998176.29 mpp[tiflash]  group by:Column#48, Column#49, Column#50, funcs:sum(Column#47)->Column#34, funcs:firstrow(Column#48)->test.orders.o_orderdate, funcs:firstrow(Column#49)->test.orders.o_shippriority, funcs:firstrow(Column#50)->test.lineitem.l_orderkey",
+		"            └─Projection_143 92822759.11 mpp[tiflash]  mul(test.lineitem.l_extendedprice, minus(1, test.lineitem.l_discount))->Column#47, test.orders.o_orderdate->Column#48, test.orders.o_shippriority->Column#49, test.lineitem.l_orderkey->Column#50",
+		"              └─Projection_118 92822759.11 mpp[tiflash]  test.orders.o_orderdate, test.orders.o_shippriority, test.lineitem.l_orderkey, test.lineitem.l_extendedprice, test.lineitem.l_discount",
+		"                └─HashJoin_117 92822759.11 mpp[tiflash]  inner join, equal:[eq(test.orders.o_orderkey, test.lineitem.l_orderkey)]",
+		"                  ├─ExchangeReceiver_53(Build) 22867441.30 mpp[tiflash]  ",
+		"                  │ └─ExchangeSender_52 22867441.30 mpp[tiflash]  ExchangeType: HashPartition, Compression: FAST, Hash Cols: [name: test.orders.o_orderkey, collate: binary]",
+		"                  │   └─Projection_51 22867441.30 mpp[tiflash]  test.orders.o_orderkey, test.orders.o_orderdate, test.orders.o_shippriority",
+		"                  │     └─HashJoin_44 22867441.30 mpp[tiflash]  inner join, equal:[eq(test.customer.c_custkey, test.orders.o_custkey)]",
+		"                  │       ├─ExchangeReceiver_48(Build) 1501762.80 mpp[tiflash]  ",
+		"                  │       │ └─ExchangeSender_47 1501762.80 mpp[tiflash]  ExchangeType: Broadcast, Compression: FAST",
+		"                  │       │   └─TableFullScan_45 1501762.80 mpp[tiflash] table:customer pushed down filter:eq(test.customer.c_mktsegment, \"AUTOMOBILE\"), keep order:false",
+		"                  │       └─TableFullScan_49(Probe) 36377904.50 mpp[tiflash] table:orders pushed down filter:lt(test.orders.o_orderdate, 1995-03-13 00:00:00.000000), keep order:false",
+		"                  └─ExchangeReceiver_57(Probe) 162359270.28 mpp[tiflash]  ",
```

Comment:
while for join's both child, anyway and currently we should keep it for now

Reply:
I have updated the comment.